### PR TITLE
Timed destruction component

### PR DESCRIPTION
**Pull request by norda97**
## Purpose & Context 
- Since Particle emitters has to be attached to an entity to spawn particles there were no way to create an entity and easily destroy it. This new component allows an entiy to be created and destroyed after a certain time. This can be used for multiple purposes, but I needed it for the spawning of emitters.

## Changes
 - Added a DestructionComponent and DestructionSystem
 - DestructionSystem is initilised in PlaySessionState and SandboxState

## Testing
How have one tested the feature to ensure it works?
- [x] Tested in Sandbox
- [x] Tested in Multiplayer with 2 clients
- [ ] Tested in Multiplayer with more than 2 clients
- [ ] Tested in Benchmark

### CODE REVIEW

File: CrazyCanvas/Include/ECS/Systems/Misc/DestructionSystem.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+	ComponentArray<DestructionComponent>* pDestructionComponents = pECS->GetComponentArray<DestructionComponent>();
+
+	for (Entity entity : m_DestructionEntities)
+	{
+		DestructionComponent& destructionComp = pDestructionComponents->GetData(entity);
+
+		if (destructionComp.Active)
+		{
+			destructionComp.TimeLeft -= deltaTime.AsSeconds();
```
Comment (norda97) [inline CrazyCanvas/Include/ECS/Systems/Misc/DestructionSystem.cpp]:
```suggestion
			destructionComp.TimeLeft -= float(deltaTime.AsSeconds());
```

File: CrazyCanvas/Include/ECS/Systems/Misc/DestructionSystem.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+				.pSubscriber = &m_DestructionEntities,
+				.ComponentAccesses =
+				{
+					{RW, DestructionComponent::Type()}
+				},
+			}
+		};
+		systemReg.Phase = 1u;
+
+		RegisterSystem(TYPE_NAME(PlayerAnimationSystem), systemReg);
```
Comment (Ceanze) [inline CrazyCanvas/Include/ECS/Systems/Misc/DestructionSystem.cpp]: Wrong type name

**Review by Ceanze — CHANGES_REQUESTED** [review]

File: CrazyCanvas/Include/ECS/Systems/Misc/DestructionSystem.cpp (hunk 2 above)
Comment (norda97) [reply CrazyCanvas/Include/ECS/Systems/Misc/DestructionSystem.cpp]: Fixed

File: CrazyCanvas/engine_config_crazycanvas.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
         1080
     ],
     "CONFIG_OPTION_FULLSCREEN": false,
-    "CONFIG_OPTION_VOLUME_MASTER": 0.22019842267036439,
+    "CONFIG_OPTION_VOLUME_MASTER": 0.0,
```
Comment (Ceanze) [inline CrazyCanvas/engine_config_crazycanvas.json:7]: very nice

**Review by Ceanze — APPROVED** [review]

**Pull request merged**